tests/integration/sequencer/test_sequencer.py

```diff
@@ -0,0 +1,280 @@
+import time
+from pathlib import Path
+
+import pytest
+
+from craft_parts import sequencer
+from craft_parts.actions import Action, ActionType
+from craft_parts.infos import ProjectInfo
+from craft_parts.parts import Part
+from craft_parts.steps import Step
+
+_pull_state_foo = textwrap.dedent(
+    """\
+    part-properties:
+      plugin: nil
+      source-subdir: ''
+      source-type: ''
+      source-depth: 0
+      source-branch: ''
+      source-tag: ''
+      source-commit: ''
+      stage-packages: []
+    project_options:
+      target_arch: amd64
+    assets:
+      stage-packages:
+      - fake-package-foo=1
+    """
+)
+
+_build_state_foo = textwrap.dedent(
+    """\
+    part-properties:
+      plugin: nil
+      after: []
+      stage-packages: []
+      disable-parallel: False
+      build-packages: []
+      organize: {}
+      build-attributes: []
+    project_options:
+      target_arch: amd64
+    assets: {}
+    """
+)
+
+_pull_state_bar = textwrap.dedent(
+    """\
+    part-properties:
+      plugin: nil
+      source-subdir: ''
+      source-type: ''
+      source-depth: 0
+      source-branch: ''
+      source-tag: ''
+      source-commit: ''
+      stage-packages: []
+    project_options:
+      target_arch: amd64
+    assets:
+      stage-packages:
+      - fake-package-bar=2
+    """
+)
+
+
+@pytest.fixture
+def pull_state(new_dir):
+    # build current state
+    Path(new_dir / "parts/foo/state").mkdir(parents=True)
+    Path(new_dir / "parts/bar/state").mkdir(parents=True)
+    Path(new_dir / "parts/foo/state/pull").write_text(_pull_state_foo)
+    Path(new_dir / "parts/bar/state/pull").write_text(_pull_state_bar)
+
+
+@pytest.mark.usefixtures("new_dir")
+class TestSequencerPlan:
+    """Verify action planning sanity."""
+
+    def test_plan_default_parts(self):
+        p1 = Part("foo", {"plugin": "nil"})
+        p2 = Part("bar", {"plugin": "nil"})
+
+        seq = sequencer.Sequencer(
+            part_list=[p1, p2],
+            project_info=ProjectInfo(),
+        )
+
+        actions = seq.plan(Step.PRIME)
+        assert actions == [
+            Action("bar", Step.PULL, action_type=ActionType.RUN),
+            Action("foo", Step.PULL, action_type=ActionType.RUN),
+            Action("bar", Step.BUILD, action_type=ActionType.RUN),
+            Action("foo", Step.BUILD, action_type=ActionType.RUN),
+            Action("bar", Step.STAGE, action_type=ActionType.RUN),
+            Action("foo", Step.STAGE, action_type=ActionType.RUN),
+            Action("bar", Step.PRIME, action_type=ActionType.RUN),
+            Action("foo", Step.PRIME, action_type=ActionType.RUN),
+        ]
+
+    def test_plan_dependencies(self):
+        p1 = Part("foo", {"plugin": "nil", "after": ["bar"]})
+        p2 = Part("bar", {"plugin": "nil"})
+
+        seq = sequencer.Sequencer(
+            part_list=[p1, p2],
+            project_info=ProjectInfo(),
+        )
+
+        # pylint: disable=line-too-long
+        # fmt: off
```

Comment:
maybe worth expanding the line length?

Reply:
Hmm, apparently there's no inline configuration to change pylint's `max-line-length`.